src/node_report.cc

```diff
@@ -809,13 +810,32 @@ static void PrintResourceUsage(std::ostream& out) {
     snprintf( buf, sizeof(buf), "%ld.%06ld", stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
     out << "\n  Kernel mode CPU: " << buf << " secs";
 #endif
-    cpu_abs = stats.ru_utime.tv_sec + 0.000001 * stats.ru_utime.tv_usec + stats.ru_stime.tv_sec + 0.000001 *  stats.ru_stime.tv_usec;
+    cpu_abs = stats.ru_utime.tv_sec + 0.000001 * stats.ru_utime.tv_usec + stats.ru_stime.tv_sec + 0.000001 * stats.ru_stime.tv_usec;
     cpu_percentage = (cpu_abs / uptime) * 100.0;
     out << "\n  Average CPU Consumption : " << cpu_percentage << "%";
     out << "\n  Filesystem activity: " << stats.ru_inblock << " reads "
         << stats.ru_oublock << " writes";
   }
-#endif
+#elif defined(__APPLE__)
+  // Currently RUSAGE_THREAD is not currently supported on Mac.
+  mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+  mach_port_t thread = pthread_mach_thread_np(pthread_self());
+  thread_basic_info thr_info;
+
+  kern_return_t rc = thread_info(thread, THREAD_BASIC_INFO, (thread_info_t) &thr_info, &count);
+
+  if (rc == KERN_SUCCESS) {
+    out << "\n\nEvent loop thread resource usage:";
+    snprintf( buf, sizeof(buf), "%d.%06d", thr_info.user_time.seconds, thr_info.user_time.microseconds);
+    out << "\n  User mode CPU: " << buf << " secs";
+    snprintf( buf, sizeof(buf), "%d.%06d", thr_info.system_time.seconds, thr_info.system_time.microseconds);
+    out << "\n  Kernel mode CPU: " << buf << " secs";
+    cpu_abs = thr_info.user_time.seconds + 0.000001 * thr_info.user_time.microseconds
+        + thr_info.system_time.seconds + 0.000001 * thr_info.system_time.microseconds;
```

Comment:
Just curious: is there a reason you multiply by 1e-6 instead of dividing by 1e6?

(Aside: I'd probably write `1e-6` instead of `0.000001` because it's more obviously correct, no need to count zeroes.)

Reply:
I wouldn't have done it like that except it meant it matched the existing conversion code above. I'd probably have done something boring like dividing by 1000000 otherwise.

The only reason I could think for the original author using 0.000001 was that possibly there's no casting between integer and floating point types in the middle of the expression (it might happen if you used 1000000 instead of 1000000.0) but I'm 100% sure how all that would have expanded.

Reply:
I think the original author was @bidipyne, in https://github.com/nodejs/node-report/commit/c9c13594e4c8a5c4a639eab44e09fee1ac739894